app.py

```
@@ -148,352 +363,645 @@
+    if 'current_folder' not in session or 'roll_number' not in session or 'name' not in session:
+        flash("Session expired or registration not started. Please start registration again.", "danger")
+        # Clear session data if it's incomplete/invalid
+        session.pop('current_folder', None)
+        session.pop('roll_number', None)
+        session.pop('name', None)
+        session.pop('is_existing_student', None)
+        print("Session data missing for capture_image.")
+        return jsonify({"status": "error", "message": "Session data missing for image capture."}), 400
+
     folder_path = session['current_folder']
+    roll_number = session['roll_number'] # Needed for potential student lookup on error
```

Ruff (0.8.2)

Local variable `roll_number` is assigned to but never used

Remove assignment to unused variable `roll_number`

(F841)